### ISSUE TITLE
rclone move ignore-existing does not ignore removing

### ISSUE THREAD

**Issue by pjw91** [rclone/rclone#5463]
#### The associated forum post URL from `https://forum.rclone.org`
https://forum.rclone.org/t/move-moveto-ignore-existing-delete-src-even-if-src-dst/11759

#### What is the problem you are having with rclone?
`move/moveto --ignore-existing` a file which exists in both source and destination folder, the destination file is left untouched, but the source file is deleted.

#### What is your rclone version (output from `rclone version`)
1.55.1

#### Which OS you are using and how many bits (e.g. Windows 7, 64 bit)
Ubuntu 20.04

#### Which cloud storage system are you using? (e.g. Google Drive)
Google Drive / Local

#### The command you were trying to run (e.g. `rclone copy /tmp remote:tmp`)
Test script (a file named both exists in both src and dst folder):
```sh
mkdir dsrc ddst                                                                                                                                                                               
echo src > dsrc/only-in-src                                                                                                                                                                   
echo dst > ddst/only-in-dst                                                                                                                                                                   
echo src > dsrc/in-both                                                                                                                                                                       
echo dst > ddst/in-both                                                                                                                                                                       
head dsrc/* ddst/*                                                                                                                                                                            
rclone move dsrc ddst --ignore-existing -vv                                                                                                                                                   
head dsrc/* ddst/*               
```

rclone move --ignore-existing src
rclone moveto --ignore-existing dst

#### A log from the command with the `-vv` flag (e.g. output from `rclone -vv copy /tmp remote:tmp`)
rclone move dsrc ddst --ignore-existing -vv
```
DEBUG : rclone: Version "v1.55.1" starting with parameters ["rclone" "move" "dsrc" "ddst" "--ignore-existing" "-vv"]
DEBUG : Creating backend with remote "dsrc"
DEBUG : fs cache: renaming cache item "dsrc" to be canonical "/dev/shm/dsrc"
DEBUG : Creating backend with remote "ddst"
DEBUG : fs cache: renaming cache item "ddst" to be canonical "/dev/shm/ddst"
DEBUG : Local file system at /dev/shm/ddst: Using server-side directory move
INFO  : Local file system at /dev/shm/ddst: Server side directory move failed - fallback to file moves: can't copy directory - destination already exists
DEBUG : Local file system at /dev/shm/ddst: Waiting for checks to finish
INFO  : only-in-src: Moved (server-side)
======BUG vvv=======
DEBUG : in-both: Destination exists, skipping
INFO  : in-both: Deleted
======BUG ^^^=======
DEBUG : Local file system at /dev/shm/ddst: Waiting for transfers to finish
INFO  : There was nothing to transfer
```

When a file exists in both src and dst, the MOVE phase is skipped, but the DELETE phase is not.
In such case, the data src/in-both is missing.

**Comment by natecollins** [comment]
I've also experienced this recently with a `move` and was surprised to find source files were deleted after having used `--ignore-existing` where existing matching filenames were present on the dest. Fortunately, I was only testing my assumptions and didn't lose any data. I had expected any source files would be ignored and left in the source dir if a matching dest file existed, similar to how rsync's `--ignore-existing` works. Said functionality would be useful to prevent accidental clobbering of files.

I did find another forum post where the topic was discussed in some depth ( [https://forum.rclone.org/t/unexpected-dangerous-behaviour-with-move-immutable-and-ignore-existing-flags/16616](https://forum.rclone.org/t/unexpected-dangerous-behaviour-with-move-immutable-and-ignore-existing-flags/16616) ), but it was auto-closed due to inactivity.

**Comment by ncw** [comment]
I think this is probably the conclusion of the thread

> Looking at the second half of your diagram, I think that it is up for discussion what `--ignore-existing` does when there is an existing file. It would be consistent to say that if a file exists on the destination then we don't delete the source. Easy to explain. 

Does this suit you @natecollins ?

**Comment by natecollins** [comment]
Indeed. Leaving the source file untouched when a matching dest file exists seems like the most expected behavior. And it avoids any potential loss of source file data.

**Comment by ncw** [comment]
@natecollins do you want to have a go at implementing this?

**Comment by natecollins** [comment]
@ncw Sure, I can give it a shot. See the above merge request. I'd appreciate if someone more familiar with the codebase could verify I've placed the changes in the proper place.

**Comment by ncw** [comment]
I have merged @natecollins solution to this to master now which means it will be in [the latest beta](https://beta.rclone.org/) in 15-30 minutes and released in v1.57

Please give it a test!

I'm going to close this - I'll re-open if necessary.

**Comment by bakugo** [comment]
I assumed the old behavior was indented and heavily relied on it, how do I get it back?
